src/libraries/System.Runtime.InteropServices.JavaScript/tests/System.Runtime.InteropServices.JavaScript.UnitTests/System/Runtime/InteropServices/JavaScript/WebWorkerTest.cs

```diff
@@ -0,0 +1,301 @@
+                }
+
+                await WebWorkerTestHelper.DisposeAsync();
+            });
+        }
+
+        [Theory, MemberData(nameof(GetTargetThreads))]
+        public async Task ThreadingTimer(Executor executor)
+        {
+            await executor.Execute(async () =>
+            {
+                TaskCompletionSource tcs = new TaskCompletionSource();
+                executor.AssertTargetThread();
+
+                using var timer = new Threading.Timer(_ =>
+                {
+                    Assert.NotEqual(1, Environment.CurrentManagedThreadId);
+                    Assert.True(Thread.CurrentThread.IsThreadPoolThread);
+                    tcs.SetResult();
+                }, null, 100, Timeout.Infinite);
+
+                await tcs.Task;
+            });
+        }
+
+        [Theory, MemberData(nameof(GetTargetThreads))]
+        public async Task JSDelay_ContinueWith(Executor executor)
+        {
+            await executor.Execute(async () =>
+            {
+                var currentTID = Environment.CurrentManagedThreadId;
+                await executor.StickyAwait(WebWorkerTestHelper.CreateDelay());
+
+                await WebWorkerTestHelper.Delay(1).ContinueWith(_ =>
+                {
+                    // continue on the context of the target JS interop
+                    switch (executor.Type)
+                    {
+                        case ExecutorType.Main:
+                            Assert.Equal(1, Environment.CurrentManagedThreadId);
+                            Assert.Equal(currentTID, Environment.CurrentManagedThreadId);
+                            Assert.False(Thread.CurrentThread.IsThreadPoolThread);
+                            break;
+                        case ExecutorType.JSWebWorker:
+                            Assert.NotEqual(1, Environment.CurrentManagedThreadId);
+                            Assert.Equal(currentTID, Environment.CurrentManagedThreadId);
+                            Assert.False(Thread.CurrentThread.IsThreadPoolThread);
+                            break;
+                        case ExecutorType.NewThread:
+                            // it will synchronously continue on the UI thread
+                            Assert.Equal(1, Environment.CurrentManagedThreadId);
+                            Assert.NotEqual(currentTID, Environment.CurrentManagedThreadId);
+                            Assert.False(Thread.CurrentThread.IsThreadPoolThread);
+                            break;
+                        case ExecutorType.ThreadPool:
+                            // it will synchronously continue on the UI thread
+                            Assert.Equal(1, Environment.CurrentManagedThreadId);
+                            Assert.NotEqual(currentTID, Environment.CurrentManagedThreadId);
+                            Assert.False(Thread.CurrentThread.IsThreadPoolThread);
+                            break;
+                    }
```

Comment:
These switch-cases are the same in `JSDelay_ContinueWith`, `JSDelay_ConfigureAwait_True`, `ManagedDelay_ConfigureAwait_True`. `ManagedYield` could use the same code as well if extracted to a private function.

```suggestion
                    AssertCorrectThread(currentTID);
```

Reply:
done, thanks